src/ATen/native/xpu/sycl/SortingKernels.h

```diff
@@ -397,10 +407,32 @@ void segmented_radix_sort_pairs_kernel(
       std::swap(keys_in_, keys_out_);
       std::swap(values_in_, values_out_);
     }
+    result_in_temp = !result_in_temp;
     begin_bit += RADIX_BITS;
     if (begin_bit >= end_bit)
       break;
   }
+
+  if (result_in_temp) {
+    auto input_calc = TrivialOffsetCalculator<2>();
+    at::detail::Array<char*, 2> data;
+    if (keys_out) {
+      data[0] = (char*)keys_out;
+      data[1] = (char*)keys_temp;
+      auto fn = InternalCopyFunctor<key_t>();
```

Comment:
I wonder if we can reduce kernel size, when we know only `bool` needs the kernel?

Reply:
Or we will have kernels for all dtpye instantiated by `key_t`, part of which are not used actually?